docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,369 @@
+
+In your cloud provider's RBAC solution, your Teleport instances need
+permissions to create, read from, and write to buckets.
+
+</Admonition>
+
+## TLS credential provisioning 
+
+High-availability Teleport deployments require an automated system to fetch
+certificates from Let's Encrypt and provision Teleport instances with them. 
+
+If you are running a single instance of the Teleport Auth Service and Proxy
+Service, you can configure this instance to fetch credentials for itself from
+Let's Encrypt using the [ACME ALPN-01
+challenge](https://letsencrypt.org/docs/challenge-types/#tls-alpn-01), where
+Teleport demonstrates that it controls the ALPN server at the HTTPS address of
+your Teleport Proxy Service.
+
+For high-availability deployments, however, it is much more efficient to use a
+separate system to request TLS credentials from Let's Encrypt and provision
+Teleport instances with them. In this case, with multiple Teleport instances
+behind a load balancer, you will need to use the [ACME
+DNS-01](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge)
+challenge to demonstrate domain name ownership to Let's Encrypt. In this
+challenge, your TLS credential provisioning system creates a DNS TXT record with
+a value expected by Let's Encrypt.
+
+In the configuration we are demonstrating in this guide, each Teleport instance
+expects TLS credentials for HTTPS to be available at the file paths
+`/etc/teleport-tls/tls.key` (private key) and `/etc/teleport-tls/tls.crt`
+(certificate).
+
+## DNS service
+
+Set up a DNS zone where you can create records for Teleport, e.g., an Amazon
+Route 53 hosted zone or Google Cloud DNS zone.
+
+You must create a DNS A record that points to the address of your load balancer. 
+
+(!docs/pages/includes/dns-app-access.mdx!)
+
+<Admonition title="Required permissions">
+
+The TLS credential provisioner we mentioned earlier needs permissions to manage
+DNS records in order to satisfy Let's Encrypt's DNS-01 challenge. If you are
+using cloud-managed solutions, you should use your cloud provider's RBAC system
+(e.g., AWS IAM) to grant a role to the Proxy Service to manage DNS records. 
+
+</Admonition>
+
+## Teleport instances
+
+Run the Teleport Auth Service and Proxy Service as a scalable group of compute
+resources, for example, a Kubernetes `Deployment`  or AWS Auto Scaling group.
+This requires running the `teleport` binary on each Kubernetes pod or virtual
+machine or in your group. In the [Configuration](#configuration) section, we
+will show you how to configure each binary for high availability.
+
+### Open ports
+
+Ensure that, on each Teleport instance, the following ports allow traffic from
+the load balancer. If you do not plan to run a service associated with a given
+port, you can plan to leave that port closed.
+
+| Port | Description |
+| - | - |
+| `3021` | Port used by Teleport Proxy Service instances to dial agents in [Proxy Peering mode](../management/operations/proxy-peering.mdx). |
```

Comment:
AFAIK this port is only for inter-proxy communications and should not be exposed publicly nor load-balanced.

Reply:
👍 I've deleted this line from each of the port tables.

Should Proxy Peering be enabled in the architecture described here? From my cursory understanding of Proxy Peering, it seems like, if agents are connecting to a Proxy Service instance via the load balancer, they don't need to be aware of individual Proxy Service instances, so Proxy Peering wouldn't be relevant here.

Reply:
I don't know if we should talk or not about proxy peering here, mainly because this is an enterprise feature and is still in preview.

However, agents are aware of individual proxy instances even through a loadbalancer 🙃 During their first connection they will discover how many proxies are connected to the cluster, and will start brute-forcing their way through the LB to have an open connection with each proxy. If they end up on a proxy they are already connected to, the proxy returns a "try again" message. [At scale, this can take some time for an agent to reconnect to all proxies.](https://github.com/gravitational/teleport/issues/13129#issuecomment-1249404054)

Without proxy peering, if a user ends up on a proxy the agent is not connected to, they will receive an error. With proxy peering, the proxy can lookup if the resource is available through another proxy. If it is, it's going to forward the connection to the agent's proxy on the proxy peering port. This capability also means the agents don't have to maintain an open connection per proxy, they can just keep 2 or 3 connections open and still be reachable.

Reply:
Thanks for clarifying!

Reply:
I agree that proxy peering is an advanced, Enterprise-only use case that probably shouldn't be talked about in the context of this guide. It's still early days and we don't have a ton of people using it yet.